From b80b37910325393d7fe6f61bf644ee76c204f3e6 Mon Sep 17 00:00:00 2001
From: Shivani Baranwal <quic_shivbara@quicinc.com>
Date: Tue, 15 Oct 2024 11:43:14 +0530
Subject: [PATCH] P2P2: Fix to check if sae_password is present

Fix the check for whether sae_password is present. Instead of checking
the static array's address which is always going to be true, verify that
the string is not empty.

Fixes: cf30af7c2f23 ("P2P2: Start P2P Client appropriately for P2P2 group")
Signed-off-by: Shivani Baranwal <quic_shivbara@quicinc.com>
---
 wpa_supplicant/p2p_supplicant.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wpa_supplicant/p2p_supplicant.c b/wpa_supplicant/p2p_supplicant.c
index ef2331277..0ec4a2e29 100644
--- a/wpa_supplicant/p2p_supplicant.c
+++ b/wpa_supplicant/p2p_supplicant.c
@@ -1890,7 +1890,7 @@ static void wpas_start_gc(struct wpa_supplicant *wpa_s,
 
 	os_memcpy(ssid->bssid, res->peer_interface_addr, ETH_ALEN);
 
-	if (res->akmp == WPA_KEY_MGMT_PASN && res->sae_password) {
+	if (res->akmp == WPA_KEY_MGMT_PASN && res->sae_password[0]) {
 		ssid->auth_alg = WPA_AUTH_ALG_SAE;
 		ssid->sae_password = os_strdup(res->sae_password);
 		if (!ssid->sae_password)
-- 
2.43.0

